app/controllers/ems_common.rb

```diff
@@ -631,36 +633,76 @@ def deleteemss
     end
   end
 
-  def call_ems_refresh(emss)
-    process_emss(emss, "refresh_ems") unless emss.empty?
+  def call_ems_task(emss, task, action_name)
+    return unless emss.any?
+    if task == "refresh_ems"
```

Comment:
A large win of this refactoring is reducing the single-method-with-many-branches-inside mess,
but here you have 2 methods call this methods only to have it run one of 2 branches.
If you moved the common part into add_success_flash_and_audit, all that remains here is 2 lines per branch — can you inline the 2 lines into `call_ems_refresh` / `call_ems_capture` and drop this method?